src/elements/earn/portfolio/v3/holdings/V3ManageProgramsModal.tsx

```diff
@@ -0,0 +1,208 @@
+import { Holding } from 'store/portfolio/v3Portfolio.types';
+import { prettifyNumber, toBigNumber } from 'utils/helperFunctions';
+import { shrinkToken } from 'utils/formulas';
+import { Button, ButtonSize, ButtonVariant } from 'components/button/Button';
+import { ContractsApi } from 'services/web3/v3/contractsApi';
+import {
+  confirmLeaveNotification,
+  genericFailedNotification,
+  rejectNotification,
+} from 'services/notifications/notifications';
+import { updatePortfolioData } from 'services/web3/v3/portfolio/helpers';
+import { ErrorCode } from 'services/web3/types';
+import { ReactComponent as IconClock } from 'assets/icons/time.svg';
+import BigNumber from 'bignumber.js';
+import dayjs from 'dayjs';
+
+interface Props {
+  holding: Holding;
+  renderButton: (onClick: () => void) => React.ReactNode;
+}
+
+export const V3ManageProgramsModal = ({ holding, renderButton }: Props) => {
+  const account = useAppSelector((state) => state.user.account);
+  const [isOpen, setIsOpen] = useState(false);
+  const [txBusy, setTxBusy] = useState(false);
+  const onClose = async () => {
+    setIsOpen(false);
+  };
+
+  const dispatch = useDispatch();
+
+  const programs = holding.programs.filter((p) =>
+    toBigNumber(p.tokenAmountWei).gt(0)
+  );
+
+  const programsMerged = programs.map((p) => ({
+    ...p,
+    ...holding.pool.programs.find((pp) => pp.id === p.id),
+  }));
+
+  const programsSorted = programsMerged.sort((a, _) => (a.isActive ? 0 : -1));
+
+  const handleLeaveClick = async (
+    id: string,
+    poolTokenAmountWei: string,
+    tokenAmountWei: string
+  ) => {
+    if (!account) {
+      console.error('handleLeaveClick because arguments are not defined');
+      return;
+    }
+    try {
+      setTxBusy(true);
+      const tx = await ContractsApi.StandardRewards.write.leave(
+        id,
+        poolTokenAmountWei
+      );
+      confirmLeaveNotification(
+        dispatch,
+        tx.hash,
+        shrinkToken(tokenAmountWei, holding.pool.decimals),
+        holding.pool.reserveToken.symbol
+      );
+      onClose();
+      setTxBusy(false);
+      await tx.wait();
+      await updatePortfolioData(dispatch);
+    } catch (e: any) {
+      console.error('handleLeaveClick', e);
+      onClose();
+      setTxBusy(false);
+      if (e.code === ErrorCode.DeniedTx) {
+        rejectNotification(dispatch);
+      } else {
+        genericFailedNotification(dispatch, 'Leave rewards failed');
+      }
+    }
+  };
+
+  return (
+    <>
+      {renderButton(() => setIsOpen(true))}
+      <ModalV3
+        title={'Manage Rewards'}
+        setIsOpen={onClose}
+        isOpen={isOpen}
+        separator
+        large
+      >
+        <div className="px-30 pb-30 pt-10">
+          <div className="text-secondary mb-48">
+            Earn BNT rewards on your bn{holding.pool.reserveToken.symbol}, there
+            are no cooldowns or withdrawal fees for adding or removing bnETH to
+            reward progrms
```

Comment:
<img width="507" alt="image" src="https://user-images.githubusercontent.com/7262066/188845716-15bbfe4e-ae89-4544-9cb9-12211ca15764.png">

Reply:
typo and wrong date